lib/formats/odata.js

```diff
@@ -418,6 +430,48 @@ const singleRowToOData = (fields, row, domain, originalUrl, query) => {
   });
 };
 
+const systemFields = new Set([
+  '__system/submissiondate',
+  '__system/updatedat',
+  '__system/submitterid',
+  '__system/submittername',
+  '__system/attachmentspresent',
+  '__system/attachmentsexpected',
+  '__system/status',
+  '__system/reviewstate',
+  '__system/deviceid',
+  '__system/edits',
+  '__system/formversion',
+]);
+
+// Validates $select query parameter including metadata properties and returns list of FormFields
+const filterFields = (formFields, select, table) => {
+  const filteredFields = new Set();
+  const fieldMap = formFields.reduce((map, field) => ({ ...map, [field.path.toLowerCase()]: field }), {});
```

Comment:
The OData name of a field can be different from the name of the field in the form: see `sanitizeOdataIdentifier()`. I'm thinking that `fieldMap` should use those OData names to map.

Reply:
Fixed it by sanitizing the path